transport/src/main/java/io/netty/channel/DefaultChannelHandlerContext.java

```diff
@@ -817,8 +803,24 @@ public ChannelFuture writeAndFlush(Object msg) {
         return writeAndFlush(msg, newPromise());
     }
 
-    private static void notifyOutboundHandlerException(Throwable cause, ChannelPromise promise) {
-        PromiseNotificationUtil.tryFailure(promise, cause, logger);
+    private void handleOutboundHandlerException(Throwable cause, boolean closeDidThrow, ChannelPromise promise) {
+        String msg = handler() + " threw an exception while handling an outbound event. This is most likely a bug";
```

Comment:
ultranit: if the handler's toString() method throws (perhaps trying to query its internal, broken, state), this would break out.   Not sure how water tight you want this to be.

Reply:
I think it should be fine ... we can adjust if needed.